File: auto-docs/executables/node/style_histogram.js
var plotly = require('plotly')('theengineear', 'o9zlr0hy6z')

var x0 = Math.random() * 500
var x1 = Math.random() * 500 + 1
var trace1 = {
  x: x0, 
  name: "control", 
  marker: {
    line: {
      color: "grey", 
      width: 0
    }, 
    color: "fuchsia", 
    opacity: 0.75
  }, 
  autobinx: false, 
  xbins: {
    start: -3.2, 
    end: 2.8, 
    size: 0.2
  }, 
  histnorm: "count", 
  type: "histogram"
};
var trace2 = {
  x: x1, 
  name: "experimental", 
  marker: {color: "rgb(255, 217, 102)"}, 
  opacity: 0.75, 
  autobinx: false, 
  xbins: {
    start: -1.8, 
    end: 4.2, 
    size: 0.2
  }, 
  type: "histogram"
};
var data = [trace1, trace2];
var layout = {
  title: "Sampled Results", 
  xaxis: {title: "Value"}, 
  yaxis: {title: "Count"}, 
  barmode: "overlay", 
  bargap: 0.25, 
  bargroupgap: 0.3, 
  bardir: "v"
};

var graph_options = {filename: "style-histogram", fileopt: "overwrite", layout: layout, auto_open: "false"}
plotly.plot(data, graph_options, function (err, msg) {
    console.log(msg);
});

File: auto-docs/executables/python/style_histogram.py
import plotly.plotly as py
from plotly.graph_objs import *
py.sign_in('TestBot', 'r1neazxo9w')
import numpy as np
x0 = np.random.randn(500)
x1 = np.random.randn(500)+1

trace1 = Histogram(
    x=x0,
    histnorm='count',
    name='control',
    autobinx=False,
    xbins=XBins(
        start=-3.2,
        end=2.8,
        size=0.2
    ),
    marker=Marker(
        color='fuchsia',
        line=Line(
            color='grey',
            width=0
        ),
        opacity=0.75
    )
)
trace2 = Histogram(
    x=x1,
    name='experimental',
    autobinx=False,
    xbins=XBins(
        start=-1.8,
        end=4.2,
        size=0.2
    ),
    marker=Marker(
        color='rgb(255, 217, 102)'
    ),
    opacity=0.75
)
data = Data([trace1, trace2])
layout = Layout(
    title='Sampled Results',
    xaxis=XAxis(
        title='Value'
    ),
    yaxis=YAxis(
        title='Count'
    ),
    barmode='overlay',
    bargap=0.25,
    bargroupgap=0.3
)
fig = Figure(data=data, layout=layout)
plot_url = py.plot(fig, filename='style-histogram', auto_open=False)